routes/course.py

```diff
@@ -49,11 +52,86 @@ def edit(user, slug=None):
     course = datamodels.get_course_by_slug(slug)
 
     if not course or not user.teaches(course):
-        raise InvalidUsage('No such course or you don\'t have permissions to edit it')
+        raise abort(404, 'No such course or you don\'t have permissions to edit it')
 
     if request.method == "POST":
-        pass
+        db = datamodels.get_session()
+        if 'year' in request.form:
+            try:
+                year = int(request.form['year'])
+            except ValueError:
+                return make_response(jsonify({'success': False, 'message': 'Year must be a number'}), 400)
+            course_year = datetime(year=year, month=12, day=31).date()
+            course.year = course_year
+        if 'skill_level' in request.form:
```

Comment:
You might want to align the request form's input names in the template to the underlying datamodel to avoid all this manual configuration, then grab just the set of keys we want the user to be able to change and send them through `**kwargs`.

Just avoid [mass assignment](https://cheatsheetseries.owasp.org/cheatsheets/Mass_Assignment_Cheat_Sheet.html) vulnerabilities.